### PR TITLE
Fix popup menus and window state, add window actions (#10155)

### PR DESCRIPTION
**Pull request by toreonify**
Full description can be found in #10155.

## Description
Theese fixes apply to X11 client in FreeRDP.

- Context menus in some applications disappear almost instantly, cannot use them.
Some application draw menus with different window flags from system context/popup menus. I've tested some combinations of flags that theese menus use. Other windows work as usual, except for dialog windows that show up as a separate window in taskbar on client side. Handling of window state flags in Windows is somewhat confusing.

- Window state is not syncronized correctly with Windows side
When window was maximized, then restored, then switched to another window in RDP or other X11 window, it maximized back. Maximized window state was not set on X11 window when Windows sent this event to client.

- Added allowed actions for windows
Client side created a X11 window without `_NET_WM_ALLOWED_ACTIONS` and allowed sending signals to maximize windows that aren't supposed to be. They can be maximized in theory, but there aren't any buttons in titlebar or in window context menu.

## Checks
- `clang-format` fixed some of the formating
- `scan-build` did not find any problems

```
scan-build: Analysis run complete.
scan-build: Removing directory '/tmp/.private/builder/scan-build-2024-05-03-060319-28834-1' because it contains no reports.
scan-build: No bugs found.
```

- `-pedantic` did not print warnings for changed code

### CODE REVIEW

**Comment by toreonify** [comment]
ffs, why I even tried to accept theese suggestions from a bot. Sorry about this mess.

File: client/X11/xf_client.c [hunk 1]
```diff
@@ -1930,6 +1930,20 @@ BOOL xf_setup_x11(xfContext* xfc)
 	xfc->WM_STATE = XInternAtom(xfc->display, "WM_STATE", False);
 	xfc->x11event = CreateFileDescriptorEvent(NULL, FALSE, FALSE, xfc->xfds, WINPR_FD_READ);
 
+	xfc->_NET_WM_ALLOWED_ACTIONS = XInternAtom(xfc->display, "_NET_WM_ALLOWED_ACTIONS", False);
```
[the remark this answers is not in the document]
Comment (akallabeth) [reply client/X11/xf_client.c:1933]: @toreonify these suggestions do make sense. while the name of the atom is `_NET_WM_ALLOWED_ACTIONS` a variable named like this collides with some naming restrictions in `C/posix/linux` so better remove the leading underscore from the variable names.

[the remark this answers is not in the document]
Comment (toreonify) [reply client/X11/xf_client.c:1933]: I don't disagree, but the bot didn't rename them in a separate file, which spawned compile errors. I'll push changes in a separate commit manually.

[the remark this answers is not in the document]
Comment (toreonify) [reply client/X11/xf_client.c]: Pushed updated files, now check completes successfully.

**Comment by akallabeth** [comment]
@freerdp-bot test

**Review by akallabeth — APPROVED** [review]
LGTM

**Comment by akallabeth** [comment]
better than before, still much to fix though.
thank you very much @toreonify

**Comment by toreonify** [comment]
> better than before, still much to fix though. thank you very much @toreonify

Thank you for taking time to review this problem. I'm glad I was helpful :)

**Pull request merged**